### ISSUE TITLE
Getting very different results than in the tutorials

### ISSUE THREAD

**Issue by e-eight** [netket/netket#1847]
I am learning how to use NetKet, and trying out the tutorials. In the first tutorial, Ising model ground state, the results I am getting are significantly different from the one in the tutorials. For example, I am not noticing any appreciable reduction in relative error using the Jastrow ansatz when compared to the mean-field ansatz.

**Mean-field** result: `Optimized energy: -19.28 ± 0.41 [σ²=84.62, R̂=1.0137] and relative error: 0.24347554146782371`
**Jastrow** result: `Optimized energy: -2.000e+01 ± nan [σ²=0.000e+00, R̂=inf], relative error: 0.21540923492538785`

I think `R̂=inf` is particularly concerning here.

My code is as follows (basically copying the code in the tutorial):
## Mean field ansatz
```
class MF(nn.Module):
    @nn.compact
    def __call__(self, x):
        lam = self.param(
            "lambda", nn.initializers.normal(), (1,), float
        )
        p = nn.log_sigmoid(lam * x)
        return 0.5 * jnp.sum(p, axis=-1)

mf_model = MF()
sampler = nk.sampler.MetropolisLocal(hi)
vstate = nk.vqs.MCState(sampler, mf_model, n_samples=512)

vstate.init_parameters()
optimizer = nk.optimizer.Sgd(learning_rate=0.05)
gs = nk.driver.VMC(H, optimizer, variational_state=vstate)
gs.run(n_iter=300)
mf_energy = vstate.expect(H)
error = abs((mf_energy.mean - eig_vals[0]) / eig_vals[0])
print(f"Optimized energy: {mf_energy} and relative error: {error}")
```
## Jastrow ansatz
```
class JasShort(nn.Module):
    @nn.compact
    def __call__(self, x):
        j1 = self.param(
            "j1", nn.initializers.normal(), (1,), float
        )
        j2 = self.param(
            "j2", nn.initializers.normal(), (1,), float
        )
        # compute nearest and next-to-nearest neighbour correlations
        corr1 = x * jnp.roll(x, -1, axis=-1)
        corr2 = x * jnp.roll(x, -2, axis=-1)

        return jnp.sum(j1 * corr1 + j2 * corr2, axis=-1)

model = JasShort()
vstate = nk.vqs.MCState(sampler, model, n_samples=1008)
optimizer = nk.optimizer.Sgd(learning_rate=0.05)
gs = nk.driver.VMC(H, optimizer, variational_state=vstate, preconditioner=nk.optimizer.SR(diag_shift=0.1))
log = nk.logging.RuntimeLog()
gs.run(n_iter=300, out=log)
print(f"Final optimized parameters: j1 = {vstate.parameters['j1']}, j2 = {vstate.parameters['j2']}")
jas_energy = vstate.expect(H)
error = abs((jas_energy.mean - eig_vals[0]) / eig_vals[0])
print(f"Optimized energy: {jas_energy}, relative error: {error}")
```

I am using NetKet 3.12.2 with Jax 0.4.28 and Python 3.11.9.

**Comment by gcarleo** [comment]
hi, can you check how we initialize the parameters in the jastrow tutorial ? I have the impression you changed the sigma of the normal initializationehich might lead to unstable optimization ?

**Comment by PhilipVinc** [comment]
> Optimized energy: -2.000e+01 ± nan [σ²=0.000e+00, R̂=inf], relative error: 0.21540923492538785

Zero variance means your samples are all identical, which usually means you have broken your ansatz in some way.
If you plot the energy you'll probably see a spike going up at some point.

A simple way to 'fix' such problems is to lower the learning rate and work your way up from there.

Also, `nn.initializers.normal()` has a gaussian with variance 1 which is usually too large for that. You should use something like `nn.initializers.normal(stddev=0.01)` for starters...

**Comment by e-eight** [comment]
This is what I am seeing in the [Jastrow part of the tutorial](https://netket.readthedocs.io/en/latest/tutorials/gs-ising.html#jastrow-ansatz), and I don't think I am doing anything different. 

![image](https://github.com/netket/netket/assets/3883241/16f37b2a-7a71-47b8-9d45-f32bba1f8e58)
![image](https://github.com/netket/netket/assets/3883241/1fc64e5c-e1f0-4b39-8399-065d8576d737)

Should there be an additional initialization step?

**Comment by PhilipVinc** [comment]
In what you're showing here Jastrow seems to converge reasonably well?

**Comment by e-eight** [comment]
This is what you have in the NetKet tutorial website. When I do the exact same thing, I get the results I mentioned above.

**Comment by PhilipVinc** [comment]
I just reran the Ising notebook (https://netket.readthedocs.io/en/latest/tutorials/gs-ising.html)  myself on a fresh collar instance and it ran fine, producing comparable results to the ones shown in the notebook.

Can you 1) show me the output of `python -m netket.tools.info` and 2) share a plot of the training curve, in order to see how it's breaking?

**Comment by PhilipVinc** [comment]
3) share a full script that runs and shows the problem. for example in your snippet above the sampler is not defined, nor the hamiltonian..

**Comment by PhilipVinc** [comment]
This runs fine for me, for example https://gist.github.com/PhilipVinc/1fff855b67fcd52a03e04e4a4d135165

**Comment by e-eight** [comment]
Hi, here are the things you requested:

1. Output of `python -m netket.tools.info` (I rebuilt my virtual environment with Python 3.11.2, but no change in output):
 ```
 ====================================================
==         NetKet Diagnostic Information          ==
====================================================
NetKet version       : 3.12.2

# Python
  - implementation   : CPython
  - version          : 3.11.2
  - distribution     : GCC 12.2.0
  - path             : /home/soham/Projects/many-body-physics/kan-quantum-states/.venv/bin/python

# Host information
  - System           : Linux-6.1.0-21-amd64-x86_64-with-glibc2.36
  - Architecture     : x86_64
  - AVX              : True
  - AVX2             : True
  - Cores            : 4

# NetKet dependencies
  - numpy            : 1.26.4
  - jaxlib           : 0.4.28
  - jax              : 0.4.28
  - equinox          : 0.11.4
  - flax             : 0.8.3
  - optax            : 0.2.2
  - numba            : 0.59.1
  - mpi4py           : unavailable
  - mpi4jax          : unavailable
  - netket           : 3.12.2

# Jax 
  - backends         : ['cpu']
    - cpu            : ['<0: cpu>']
```
2. Plot of training curve:
![ising-ground-state-mf-jastrow-comparison](https://github.com/netket/netket/assets/3883241/9081313f-baa1-477a-82b4-c7fa889047b9)
3. Full script: https://gist.github.com/e-eight/c71a8468e18d9ec23e10116821483242

**Comment by gcarleo** [comment]
you set Gamma=1 instead of Gamma=-1. This apparently small changes makes the ground state non positive definitr, but you are still working with a positive definite jastrow ansatz (since you are taking the parameters to be float and not complex) thus you are unable to learn the wave function

**Comment by e-eight** [comment]
Oh!! Thanks for catching that. Very silly error on my part. It makes sense that it won't be able to learn the wave function.
